runtime/bcutil/ROMClassWriter.cpp

```diff
@@ -1655,6 +1657,77 @@ ROMClassWriter::writeSourceDebugExtension(Cursor *cursor)
 	}
 }
 
+/* 
+ * Records in the ROM class has the following layout:
+ * 4 bytes for the number of record components in the class
+ * for each record component:
+ * 	J9ROMRecordComponentShape
+ * 	4 bytes SRP to record component's generic signature if the component has a signature annotation.
+ * 	4 bytes length of annotation data in bytes, followed by annotation data. Omitted if there are no annotations.
+ * 	4 bytes length of type annotation data in bytes followed by type annotation data.  Omitted if there are no type annotations.
```

Comment:
Can you clarify the layout for a record that has a generic signature but no other annotations?
Is it:
```
4 bytes SRP
```
or:
```
4 bytes SRP
0
0
```?

Reply:
I assume the first and the bits in the attributeFlags are checked to determine what data is there?

Reply:
It would be the first layout you mentioned. I will clarify that in the comment block.